react/features/base/settings/middleware.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import { setAudioOnly } from '../conference';
+import { getLocalParticipant, participantUpdated } from '../participants';
+import { MiddlewareRegistry } from '../redux';
+
+import { SETTINGS_UPDATED } from './actionTypes';
+
+/**
+ * The middleware of the feature base/settings. Distributes changes to the state
+ * of base/settings to the states of other features computed from the state of
+ * base/settings.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    const result = next(action);
+
+    switch (action.type) {
+    case SETTINGS_UPDATED:
+        _maybeSetAudioOnly(store, action);
+        _updateLocalParticipant(store, action);
+    }
+
+    return result;
+});
+
+/**
+ * Maps the settings field names to participant names where they don't match.
+ * Currently there is only one such field, but may be extended in the future.
+ *
+ * @private
+ * @param {string} settingsField - The name of the settings field to map.
+ * @returns {string}
+ */
+function _mapSettingsFieldToParticipant(settingsField) {
```

Comment:
Now that we are moving the profile to settings, can we align these and use settings.name instead? It feels weird to introduce this function when it looks like we can store it as "name" now. WDYT?

Reply:
displayName is the identifier that we use all across the app (even in the public API) and it's also persisted that way so I wouldn't change that. Changing participant.name to displayName seems to be a more viable thing, but that would include quite a lot of work (I don't even now how to discover all its references) and I have no idea what it could break.

Reply:
Maybe a larger refactor later where we also change the jwt field names and everything to be aligned? Just thinking out loud. Let's not stretch this now :)

Reply:
Ok.